File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    const sum = brown + yellow;
    let x = 0;
    let y = 0;
    
    for(let i=3; i<=sum; i++) {
        if(sum%i === 0 && (i+sum/i)*2-4 === brown) {
            x = i;
            y = sum/i;
        }
    }
    
    return [x, y]
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    s, w, h = brown + yellow, 0, 0

    for i in range(3, s):
        if s % i == 0 and (i + s/i) * 2 - 4 == brown:
            w, h = i, s/i

    return [w, h]